pkg/engine/engine_test.go

```diff
@@ -445,4 +445,33 @@ func TestAlterFuncMap(t *testing.T) {
 		t.Errorf("Expected %q, got %q (%v)", expectNum, gotNum, outReq)
 	}
 
+	tplChart := &chart.Chart{
+		Metadata: &chart.Metadata{Name: "TplFunction"},
+		Templates: []*chart.Template{
+			{Name: "templates/base", Data: []byte(`Evaluate tpl {{tpl "Value: {{ .Values.value}}" .}}`)},
```

Comment:
This looks great!

Can you also test a template that uses template functions (like `quote`) and defined templates (`{{ define "foo" }}`)? Something like `tpl "Value: {{ .Values.value | quote }}"` and `tpl "{{ include 'foo' }}"`.

It looks like those should both work. But I think it's worth having a safety net in the tests. We need to be sure that now and in the future those two items are available within a nested template.